packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useState } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+
+import { useAwaitQuery } from '../../hooks/useAwaitQuery';
+import { useAppContext } from '../../context/app';
+import { useUserContext } from '../../context/user';
+import { useCartContext } from '../../context/cart';
+
+export const useCheckoutPage = props => {
+    const { createCartMutation, getCartDetailsQuery } = props;
+
+    const [, { toggleDrawer }] = useAppContext();
+    const [{ isSignedIn }] = useUserContext();
+    const [
+        { isEmpty },
+        { createCart, getCartDetails, removeCart }
+    ] = useCartContext();
+
+    const [fetchCartId] = useMutation(createCartMutation);
+    const fetchCartDetails = useAwaitQuery(getCartDetailsQuery);
```

Comment:
Even though we are using the redux actions for create cart (as there isn't a better alternative now) I think that fetching data for the checkout page should be done identically to how the cart page does it.

Create a lazy query and call it in an effect whenever the `cartId` changes (so long as there is a cartId).

See https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/talons/CartPage/useCartPage.js#L13-L34

Reply:
Like I mentioned previously, I'll leave this to the individual working on the ticket. The scope of this PR is to set up a basic state and UI.